### PR TITLE
Add Python API for keyword spotting

### PR DESCRIPTION
**Pull request by pkufool**

### CODE REVIEW

File: python-api-examples/keyword-spotter-from-microphone.py [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+        between each other.
+        """,
+    )
+
+    parser.add_argument(
+        "--keywords-file",
+        type=str,
+        default="",
+        help="""
+        The file containing keywords, one words/phrases per line, and for each
```
Comment (csukuangfj) [inline python-api-examples/keyword-spotter-from-microphone.py:102]:
```suggestion
        The file containing keywords, one word/phrase per line, and for each
```

File: python-api-examples/keyword-spotter-from-microphone.py [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+        help="""The number of tailing blanks a keyword should be followed. Setting
+        to a larger value (e.g. 8) when your keywords has overlapping tokens
+        between each other.
+        """,
+    )
+
+    parser.add_argument(
+        "--keywords-file",
+        type=str,
+        default="",
```
Comment (csukuangfj) [inline python-api-examples/keyword-spotter-from-microphone.py]:
```suggestion
```

It cannot be empty.

File: python-api-examples/keyword-spotter-from-microphone.py [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+    parser.add_argument(
+        "sound_files",
+        type=str,
+        nargs="+",
+        help="The input sound file(s) to decode. Each file must be of WAVE"
+        "format with a single channel, and each sample has 16-bit, "
+        "i.e., int16_t. "
+        "The sample rate of the file can be arbitrary and does not need to "
+        "be 16 kHz",
+    )
```
Comment (csukuangfj) [inline python-api-examples/keyword-spotter-from-microphone.py]:
```suggestion
```

File: python-api-examples/keyword-spotter.py [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+        help="""The number of tailing blanks a keyword should be followed. Setting
+        to a larger value (e.g. 8) when your keywords has overlapping tokens
+        between each other.
+        """,
+    )
+
+    parser.add_argument(
+        "--keywords-file",
+        type=str,
+        default="",
```
Comment (csukuangfj) [inline python-api-examples/keyword-spotter.py]:
```suggestion
```

File: sherpa-onnx/python/sherpa_onnx/keyword_spotter.py [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+
+from _sherpa_onnx import KeywordSpotter as _KeywordSpotter
+
+
+def _assert_file_exists(f: str):
+    assert Path(f).is_file(), f"{f} does not exist"
+
+
+class KeywordSpotter(object):
+    """A class for streaming speech recognition.
```
Comment (csukuangfj) [inline sherpa-onnx/python/sherpa_onnx/keyword_spotter.py]:
```suggestion
    """A class for streaming keyword spotting.
```

File: sherpa-onnx/python/sherpa_onnx/keyword_spotter.py [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+            provider=provider,
+        )
+
+        feat_config = FeatureExtractorConfig(
+            sampling_rate=sample_rate,
+            feature_dim=feature_dim,
+        )
+
+        keywords_spotter_config = KeywordSpotterConfig(
+            feat_config,
```
Comment (csukuangfj) [inline sherpa-onnx/python/sherpa_onnx/keyword_spotter.py]: Please use 
```
key=value
```
for this function.

File: sherpa-onnx/python/sherpa_onnx/keyword_spotter.py [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+        self,
+        tokens: str,
+        encoder: str,
+        decoder: str,
+        joiner: str,
+        num_threads: int = 2,
+        sample_rate: float = 16000,
+        feature_dim: int = 80,
+        max_active_paths: int = 4,
+        keywords_file: str = "",
```
Comment (csukuangfj) [inline sherpa-onnx/python/sherpa_onnx/keyword_spotter.py]: Is `keywords_file` allowed to be empty?

I see that you check its existence explicitly in the example.

If it is not allowed to be empty, please don't use a default empty value for it.

**Review by csukuangfj — CHANGES_REQUESTED** [review]

**Comment by lonngxiang** [comment]
@pkufool  非常感谢；这里有个单词拼写错了https://github.com/k2-fsa/sherpa-onnx/issues/621

**Pull request merged**